protocol-units/mempool/move-rocks/src/lib.rs

```diff
@@ -34,27 +36,19 @@ impl RocksdbMempool {
 		Ok(RocksdbMempool { db: Arc::new(RwLock::new(db)) })
 	}
 
-
 	pub fn construct_mempool_transaction_key(transaction: &MempoolTransaction) -> String {
 		// Pre-allocate a string with the required capacity
 		let mut key = String::with_capacity(32 + 1 + 32 + 1 + 32);
-	
-		// Pad to 32 characters for slot_seconds and append to key
-		key.push_str(&format!("{:032}", transaction.timestamp));
-		key.push(':');
-	
-		// Pad to 32 characters for sequence number and append to key
-		key.push_str(&format!("{:032}", transaction.transaction.sequence_number));
-		key.push(':');
-	
-		// Assuming transaction.transaction.id() returns a hex string of length 32
-		let transaction_id_hex = transaction.transaction.id(); // This should be a String of hex characters
-		key.push_str(&transaction_id_hex.to_string().as_str());
-	
+		// Write key components. The numbers are zero-padded to 32 characters.
+		key.write_fmt(format_args!(
+			"{:032}:{:032}:{}",
+			transaction.timestamp,
+			transaction.transaction.sequence_number,
+			transaction.transaction.id(),
+		))
+		.unwrap();
```

Comment:
Note that using a binary key instead of a stringized decimal-hex one would improve performance much more than these micro-optimizations.

Reply:
Feel free to make this change.